components/style/values/generics/basic_shape.rs

```diff
@@ -123,3 +126,105 @@ impl<L: ToComputedValue> ToComputedValue for ShapeRadius<L> {
         }
     }
 }
+
+// https://drafts.csswg.org/css-shapes/#typedef-fill-rule
+// NOTE: Basic shapes spec says that these are the only two values, however
+// https://www.w3.org/TR/SVG/painting.html#FillRuleProperty
+// says that it can also be `inherit`
+define_css_keyword_enum!(FillRule:
+    "nonzero" => NonZero,
+    "evenodd" => EvenOdd
+);
+
+impl ComputedValueAsSpecified for FillRule {}
+
+impl Default for FillRule {
+    #[inline]
+    fn default() -> Self { FillRule::NonZero }
+}
+
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+#[allow(missing_docs)]
+/// https://drafts.csswg.org/css-shapes/#funcdef-polygon
+pub struct Polygon<L> {
+    pub fill: FillRule,
+    pub coordinates: Vec<(L, L)>,
+}
+
+impl<L: Parse> Polygon<L> {
+    #[allow(missing_docs)]
+    pub fn parse_function_arguments(context: &ParserContext, input: &mut Parser) -> Result<Self, ()> {
+        let fill = input.try(|i| {
+            let fill = FillRule::parse(i);
+            i.expect_comma()?;      // only eat the comma if there is something before it
```

Comment:
What guarantees that input doesn't start with a comma? That'd be incorrect in that case wouldn't it? Well, it'd rewind if there isn't anything, so perhaps it's fine, but I find it more confusing than:

```rust
let fill = FillRule::parse(i)?;
i.expect_comma()?;
Ok(fill)
```